test/model/autocomplete_test.dart

```diff
@@ -349,4 +320,131 @@ void main() {
+      await store.addUsers(users);
+    }
+
+    group('MentionAutocompleteView.compareByDms', () {
+      test('has DMs with userA and userB, latest with userA, prioritizes userA', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [1],    maxMessageId: 200),
+          RecentDmConversation(userIds: [1, 2], maxMessageId: 100),
+        ]);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).isNegative();
+      });
+
+      test('has DMs with userA and userB, latest with userB, prioritizes userB', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [2],    maxMessageId: 200),
+          RecentDmConversation(userIds: [1, 2], maxMessageId: 100),
+        ]);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).isGreaterThan(0);
+      });
+
+      test('has DMs with userA and userB, equally recent, prioritizes neither', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [1, 2], maxMessageId: 100),
+        ]);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).equals(0);
+      });
+
+      test('has DMs with userA but not userB, prioritizes userA', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [1], maxMessageId: 100),
+        ]);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).isNegative();
+      });
+
+      test('has DMs with userB but not userA, prioritizes userB', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [2], maxMessageId: 100),
+        ]);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).isGreaterThan(0);
+      });
+
+      test('doesn\'t have DMs with userA or userB, prioritizes neither', () async {
+        await prepare(dmConversations: []);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).equals(0);
+      });
+    });
+
+    test('autocomplete suggests relevant users in the following order: '
+      '1. Users most recent in the DM conversations', () async {
+      final users = [
```

Comment:
nit: indent continuation line further to avoid looking like part of the body:
```suggestion
    test('autocomplete suggests relevant users in the following order: '
        '1. Users most recent in the DM conversations', () async {
      final users = [
```

But taking a step back, if the test name feels like it needs to spill onto a second line, that's a sign that it's too long for a name :slightly_smiling_face:

Reply:
This test's name will take more and more lines once we add other criteria. I thought about shrinking its name, but as this test is about exercising the different criteria users are suggested based on, and to provide the reader with enough context, I couldn't come up with a shorter name. I would appreciate any clue from your side!🙂

Reply:
For someone reading the source code, information is just as visible if it's in a comment as in the test name. So the most direct fix is to move the extra information from the name to a comment, e.g. replacing:

```dart
    group('autocomplete suggests relevant users in the following order: '
        '1. Users most recent in the DM conversations', () {
      Future<void> checkResultsIn(Narrow narrow, {required List<int> expected}) async {
```

with:
```dart
    group('autocomplete suggests relevant users in the intended order', () {
      // The order should be:
      // 1. Users most recent in the DM conversations

      Future<void> checkResultsIn(Narrow narrow, {required List<int> expected}) async {
```

The role of a test name is to identify the test case. It appears in places like the status line that `flutter test` shows you, with the name of a currently-running test, or in a heading line when a test fails; and it can be used with `flutter test --name` to pick specific tests to run. If the name gets much longer than a line, it's not really readable in those formats anyway.

For example here's the names in your current revision:
![image](https://github.com/zulip/zulip-flutter/assets/28173/198377c8-207b-4363-aa01-746815f86613)

(Usually I'd post command output like this as a code block, but I think for this discussion it's helpful to see how it looks on the screen.)

And here it is with the tweak above:
![image](https://github.com/zulip/zulip-flutter/assets/28173/2e8000e1-a503-4dba-aa23-df2dfc4f4267)

Still kind of long, but getting more manageable. And I think the details of the order don't come across particularly helpfully in any case.

Reply:
Separately: if there's a test where you feel like it takes more than a line to name what it's about, that's also a sign that the test itself may be combining too many different things. I suspect that's also something that's going on here, but it's relatively mild at this stage and will become more important in #692 where there are more ingredients in the sorting.

When an individual test case is testing too many ideas at once, it tends to make it so that when something breaks the test, it's hard to tell what the test intended; and as a result it's hard to tell whether the test just needs to be updated to expect the new output, or the test is telling you something you broke in the actual application code and should fix. The main point of a test case is to tell you when some future change accidentally broke the functionality the test is testing; so when that becomes hard to tell, it largely defeats the purpose of the test.